docs/integrations/ml-dist.rst

```diff
@@ -0,0 +1,512 @@
+
+1. Retrieve data from CrateDB.
+2. Show extracts of how the data looks.
+3. Prepare the data and build a machine learning model that will predict
+   if the machine will fail in a specified time window in the future.
+4. Save the previously built model on AWS S3.
+5. Retrieve the model from AWS S3, predict sample data and insert the results
+   into the CrateDB table ``ml.pump_sensors_predictions``.
+
+Prerequisites
+-------------
+
+- `A running CrateDB cluster`_
+- `A AWS S3 storage bucket`_
+- A AWS config file at ``~/.aws/config`` containing:
+
+.. code-block:: python
+
+    [default]
+    aws_access_key_id=<key>
+    aws_secret_access_key=<access_key>
+
+- `Python`_
+    - The `tensorflow`_ library
+    - The `numpy`_ library
+    - The `sklearn`_ library
+    - The `pandas`_ library
+    - The `psycopg2`_ library
+    - The `boto3`_ library
+    - The `joblib`_ library
+    - The `matplotlib`_ library
+
+
+CrateDB
+-------
+
+In this first step we need to make sure that we create the table that holds the pump_sensor
+data set as well as the table that will be saving the predictions:
+
+.. code-block:: sql
+
+    CREATE TABLE "ml"."pump_sensors"
+    (
+       "id" INTEGER,
+       "timestamp" TEXT,
+       "sensor_00" REAL,
+       "sensor_01" REAL,
+       "sensor_02" REAL,
+       "sensor_03" REAL,
+       "sensor_04" REAL,
+       "sensor_05" REAL,
+       "sensor_06" REAL,
+       "sensor_07" REAL,
+       "sensor_08" REAL,
+       "sensor_09" REAL,
+       "sensor_10" REAL,
+       "sensor_11" REAL,
+       "sensor_12" REAL,
+       "sensor_13" REAL,
+       "sensor_14" REAL,
+       "sensor_15" REAL,
+       "sensor_16" REAL,
+       "sensor_17" REAL,
+       "sensor_18" REAL,
+       "sensor_19" REAL,
+       "sensor_20" REAL,
+       "sensor_21" REAL,
+       "sensor_22" REAL,
+       "sensor_23" REAL,
+       "sensor_24" REAL,
+       "sensor_25" REAL,
+       "sensor_26" REAL,
+       "sensor_27" REAL,
+       "sensor_28" REAL,
+       "sensor_29" REAL,
+       "sensor_30" REAL,
+       "sensor_31" REAL,
+       "sensor_32" REAL,
+       "sensor_33" REAL,
+       "sensor_34" REAL,
+       "sensor_35" REAL,
+       "sensor_36" REAL,
+       "sensor_37" REAL,
+       "sensor_38" REAL,
+       "sensor_39" REAL,
+       "sensor_40" REAL,
+       "sensor_41" REAL,
+       "sensor_42" REAL,
+       "sensor_43" REAL,
+       "sensor_44" REAL,
+       "sensor_45" REAL,
+       "sensor_46" REAL,
+       "sensor_47" REAL,
+       "sensor_48" REAL,
+       "sensor_49" REAL,
+       "sensor_50" REAL,
+       "sensor_51" REAL,
+       "machine_status" TEXT
+    );
+
+    CREATE TABLE "ml"."pump_sensors_predictions"
+    (
+       "prediction" REAL,
+       "ts" TIMESTAMP WITH TIME ZONE,
+       "model_name" TEXT
+    );
+
+After creating the tables we will start with looking at our data before we will
+insert data to our table, since the data might not be in a state where it can
+be easily inserted into our CrateDB instance.
```

Comment:
I would rephrase this to something like:

Before we insert any data into the ``pump_sensors`` table, we should first inspect it in case it needs processing before it can be successfully inserted.